src/snowflake/connector/connection.py

```diff
@@ -1749,6 +1747,12 @@ def _cache_query_status(self, sf_qid: str, status_ret: QueryStatus) -> None:
             )  # Prevent KeyError when multiple threads try to remove the same query id
             self._done_async_sfqids[sf_qid] = None
 
+    def _close_at_exit(self):
+        try:
```

Comment:
Maybe use [contextlib.suppress()](https://docs.python.org/3/library/contextlib.html#contextlib.suppress)?

Reply:
this is elegant